### PR TITLE
changelogs: categorize deprecations under deprecated_features

### PR DESCRIPTION
**Pull request by gotmax23**
##### SUMMARY
changelogs: categorize deprecations under deprecated_features

These should be put under deprecated_features so they show up properly in the generated changelog.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME
changelogs

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: changelogs/changelog.yaml [hunk 1]
```diff
@@ -158,7 +158,7 @@ releases:
     release_date: '2024-08-22'
   3.10.1:
     changes:
-      breaking_changes:
+      deprecated_features:
       - mysql_user - the ``user`` alias of the ``name`` argument has been deprecated
```
Comment (felixfontein) [inline changelogs/changelog.yaml:162]: Also 3.10.1 is a bugfix release, it should **not** have deprecations. See the semantic versioning spec: https://semver.org/

**Review by felixfontein — APPROVED** [review]

File: changelogs/changelog.yaml (hunk 1 above)
Comment (laurent-indermuehle) [reply changelogs/changelog.yaml:162]: @felixfontein , I think it was not a deprecation. It's a deprecation warning. The removal will happen in 5.0.
So the version was correct, right?

Comment (felixfontein) [reply changelogs/changelog.yaml:162]: When you mark something as deprecated and add a waring, that's a deprecation, and according to https://semver.org/ must happen in a minor or major release, not in a bugfix release.

(The actual removal of the functionality must happen in a major release.)

Comment (Andersson007) [reply changelogs/changelog.yaml:162]: cool, now we know it for future cases, thanks for the info!

**Comment by Andersson007** [comment]
@gotmax23 thanks for fixing this!
@laurent-indermuehle @felixfontein thanks for reviewing!

**Pull request merged**